File: lib/broadcast.rb
require 'pusher'

class Broadcast
  
  Pusher.app_id = '990'
  Pusher.key = '064cfff6a7f7e44b07ae'
  Pusher.secret = 'bea582cd929821f3f0f0'
  
  class << self  
    def user_online(domain, user)
      # avatar = user.profile.avatar
      # Pusher['wave'].trigger('user-online', { :full_name => user.full_name, :avatar => { :id => avatar.try(:id), :file_name => avatar.try(:image_file_name) }})            
      Pusher['wave'].trigger('user-online', { :full_name => user.full_name })            
    end
  
    def user_offline(domain, user)
      Pusher['wave'].trigger('user-offline', { :full_name => user.full_name })
    end
  
    def user_create(domain, user)
      Pusher['wave'].trigger('user-register', { :full_name => user.full_name })      
    end  
  end
end

File: public/javascripts/broadcast.js
var PusherEvent = {
  log: function(message, avatar) {
    message = '<p>' + message
    if (avatar) {
      // message = message + avatar.id + avatar.image_file_name
    }
    message = message + '</p>'
    $('#events').prepend(message);
  }
};

jQuery(document).ready(function($) {
  var server = new Pusher('064cfff6a7f7e44b07ae', 'wave');
  server.bind('user-online', function(user) {
    var avatar_url;
    // if (user.avatar !== undefined) {
      // avatar_url = '/system/images/' + avatar.id + '/thumb/' + avatar.file_name
    // }
    PusherEvent.log(user.full_name + ' arrived', avatar);
  });
  server.bind('user-register', function(user) {
    PusherEvent.log(user.full_name + ' registered!');
  });
  server.bind('lurker-online', function(user) {
    PusherEvent.log('A lurker arrived');
  });
  server.bind('user-offline', function(user) {
    PusherEvent.log(user.full_name + ' logged out');
  });
});